parcels/grid.py

```diff
@@ -78,7 +79,16 @@ def from_netcdf(cls, filenames, variables, dimensions, **kwargs):
             fields[var] = Field.from_netcdf(var, dimensions, dsets, **kwargs)
         u = fields.pop('U')
         v = fields.pop('V')
-        return cls(u, v, u.depth, u.time, fields=fields)
+
+        # assign time_origin if the time dimension has units and calendar
+        try:
+            time_units = dsets[0][dimensions['time']].units
+            calendar = dsets[0][dimensions['time']].calendar
+            time_origin = num2date(0, time_units, calendar)
```

Comment:
Nice! Looks good, it appears that it makes the calculation for me now when `time_origin` is being assigned in `Field`. Cool, one less thing to worry about! 😎